### PR TITLE
Add Configuration with specific modules.

### PR DESCRIPTION
**Pull request by boaks**
Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

File: element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java [hunk 1]
```diff
@@ -1507,12 +1542,16 @@ public static Configuration createStandardWithoutFile() {
...
+	 * Apply all {@link ModuleDefinitionsProvider} of registered modules.
 	 * 
 	 * @param inStream input stream to read properties.
 	 * @return the standard configuration
+<<<<<<< cur
 	 * @throws NullPointerException if the in stream is {@code null}.
 	 * @see #addModule(String, DefinitionsProvider)
+=======
+	 * @see #addModule(ModuleDefinitionsProvider)
+>>>>>>> dbed5f2 Add Configuration with specific modules.
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java:1554]: It seems there is some conflict artifact remaining.

File: element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java [hunk 2]
```diff
@@ -1550,16 +1593,20 @@ public static Configuration createFromStream(InputStream inStream, DefinitionsPr
...
 	 * {@link Configuration#createStandardFromStream(InputStream)}.
 	 * 
 	 * @param file the configuration file
 	 * @return the standard configuration
+<<<<<<< cur
 	 * @throws NullPointerException if the file is {@code null}.
 	 * @see #addModule(String, DefinitionsProvider)
+=======
+	 * @see #addModule(ModuleDefinitionsProvider)
+>>>>>>> dbed5f2 Add Configuration with specific modules.
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java:1609]: same here

File: element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java [hunk 3]
```diff
@@ -1582,8 +1629,12 @@ public static Configuration createStandardWithFile(File file) {
 	 * @param header The header to write to the top of the file.
 	 * @param customProvider custom definitions handler. May be {@code null}.
 	 * @return the configuration
+<<<<<<< cur
 	 * @throws NullPointerException if the file or header is {@code null}.
 	 * @see #addModule(String, DefinitionsProvider)
+=======
+	 * @see #addModule(ModuleDefinitionsProvider)
+>>>>>>> dbed5f2 Add Configuration with specific modules.
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java:1637]: same here

File: element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java [hunk 4]
```diff
@@ -1672,8 +1746,13 @@ public void load(final InputStream inStream) throws IOException {
 	 * Apply conversion defined by that {@link DocumentedDefinition}s.
 	 * 
 	 * @param properties properties to convert and add
+<<<<<<< cur
 	 * @throws NullPointerException if properties is {@code null}.
 	 * @see #addModule(String, DefinitionsProvider)
+=======
+	 * @see #addModule(ModuleDefinitionsProvider)
+	 * @see #Configuration(ModuleDefinitionsProvider...)
+>>>>>>> dbed5f2 Add Configuration with specific modules.
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java:1755]: same here

File: element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java [hunk 5]
```diff
@@ -1704,8 +1783,13 @@ public void add(Properties properties) {
 	 * {@code true}.
 	 * 
 	 * @param dictionary dictionary to convert and add
+<<<<<<< cur
 	 * @throws NullPointerException if dictionary is {@code null}.
 	 * @see #addModule(String, DefinitionsProvider)
+=======
+	 * @see #addModule(ModuleDefinitionsProvider)
+	 * @see #Configuration(ModuleDefinitionsProvider...)
+>>>>>>> dbed5f2 Add Configuration with specific modules.
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java:1792]: same here

File: element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java (hunk 2 above)
Comment (boaks) [reply element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java:1609]: yep. I rebased and fixed it.  Let me test it, before I push it.

File: element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java (hunk 5 above)
Comment (boaks) [reply element-connector/src/main/java/org/eclipse/californium/elements/config/Configuration.java:1792]: And some more ...

**Comment by boaks** [comment]
Rebased, corrected and pushed ...

**Pull request merged**